### DISCUSSION TITLE
Planetscale workflow with Vercel

### DISCUSSION THREAD

**Discussion by jcohenho** [planetscale/discussion#187] [FAQs]
I'm noticing that the Planetscale integration with Vercel sets both preview and production environment variables to the same database credentials. How do you propose we work with preview apps to test new features and data changes without polluting production data? It's easy in development via the CLI proxy to connect to a different branch and work with test data, but I'm not sure how to replicate that with preview apps.

**Comment by tbarn** [comment]
You are correct in that it sets both of those to production. Right now the integration isn't smart enough to know what PlanetScale branch you want to test your preview app on, so you will need to update it manually. This is great feedback though, thank you!

**Reply by jcohenho** [reply]
My question has some overlap with ongoing discussions [here](https://github.com/planetscale/discussion/discussions/175), specifically this part: 
> there's no convenient way to pair isolated database environments with every ephemeral Vercel preview deployment, and I think PlanetScale because of its branching feature is in a great position to take that spot. 

As Vercel automatically spins up a new preview app, it would be amazing to have PlanetScale automatically create a branch that pairs with it.  I can see how this might be confusing when trying to create a PR with a schema change and relevant DR, where it needs to point to a specific branch, but maybe there's a smart way to handle that difference.

**Reply by KevinMind** [reply]
Totally agree here. I'm actually very surprised / disappointed this is not the default behaviour. With vercel and planetscale heavily invested in both serverless and the deployment per branch model ... why doesn't this just work?

**Reply by kempsterrrr** [reply]
This behaviour would dramatically improve our workflows 🙏🏻

**Reply by nmajor** [reply]
+1 for this feature
